packages/eslint-plugin/src/rules/require-object-type-annotations.ts

```diff
@@ -0,0 +1,55 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import type { RuleListener } from '@typescript-eslint/utils/src/ts-eslint';
+import type * as ts from 'typescript';
+
+import * as util from '../util';
+
+export default util.createRule({
+  name: 'require-object-type-annotations',
+  defaultOptions: [],
+  meta: {
+    docs: {
+      description: '',
+      requiresTypeChecking: true,
+      recommended: false,
+    },
+    messages: {
+      forbidden: 'Object is missing type annotation.',
+    },
+    schema: [],
+    type: 'problem',
+  },
+  create: (context): RuleListener => {
+    const listener = (esNode: TSESTree.ObjectExpression): void => {
```

Comment:
nit - just inline this with the selector to save adding the annotations

Reply:
0b79c5edb4a0cb99735c0f1cbc17d1d872a23edf